### PR TITLE
[feature](lakehouse) use parquet bloom for row group pruning

### PR DESCRIPTION
**Pull request by Nitin-Kashyap**
## Proposed changes

Added support for reading Parquet Blooms and use the same for RowGroup Filtering.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/parquet/parquet_bloom.h [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    // Maximum Bloom filter size, it sets to HDFS default block size 128MB
+    // This value will be reconsidered when implementing Bloom filter producer.
+    static constexpr uint32_t kMaximumBloomFilterBytes = 128 * 1024 * 1024;
+
+    /// Determine whether an element exist in set or not.
+    ///
+    /// @param hash the element to contain.
+    /// @return false if value is definitely not in set, and true means PROBABLY
+    /// in set.
+    virtual bool FindHash(uint64_t hash) const = 0;
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/parquet_bloom.h:36]:
```suggestion
    virtual bool find_hash(uint64_t hash) const = 0;
```
Same as other methods' name

**Comment by morningman** [comment]
Does this code referenced from somewhere else?
Because the code style is not same as Doris.
If yes, please add reference link in code

**Comment by Nitin-Kashyap** [comment]
> Does this code referenced from somewhere else? Because the code style is not same as Doris. If yes, please add reference link in code
yes, its refered from the parquet code flow.. I'll add that reference

File: be/src/vec/exec/format/parquet/parquet_bloom_reader.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+public:
+    virtual ~BloomFilterReader() = default;
+
+    /// \brief Create a BloomFilterReader instance.
+    /// \returns a BloomFilterReader instance.
+    /// WARNING: The returned BloomFilterReader references to all the input parameters, so
+    /// it must not outlive all of the input parameters. Usually these input parameters
+    /// come from the same ParquetFileReader object, so it must not outlive the reader
+    /// that creates this BloomFilterReader.
+    static std::unique_ptr<BloomFilterReader> Make(io::FileReaderSPtr file_reader,
```
[the remark this answers is not in the document]
Comment (Nitin-Kashyap) [reply be/src/vec/exec/format/parquet/parquet_bloom_reader.h:59]: fixed

File: be/src/vec/exec/format/parquet/parquet_bloom_reader.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                                                   io::IOContext* io_context);
+
+    /// \brief Get the bloom filter reader of a specific row group.
+    /// \param[in] i row group ordinal to get bloom filter reader.
+    /// \returns RowGroupBloomFilterReader of the specified row group. A nullptr may or may
+    ///          not be returned if the bloom filter for the row group is unavailable. It
+    ///          is the caller's responsibility to check the return value of follow-up calls
+    ///          to the RowGroupBloomFilterReader.
+    /// \throws ParquetException if the index is out of bound.
+    virtual std::shared_ptr<RowGroupBloomFilterReader> RowGroup(int i) = 0;
```
[the remark this answers is not in the document]
Comment (Nitin-Kashyap) [reply be/src/vec/exec/format/parquet/parquet_bloom_reader.h:70]: fixed

File: be/src/vec/exec/format/parquet/parquet_bloom_reader.h [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    virtual std::shared_ptr<RowGroupBloomFilterReader> RowGroup(
```
[the remark this answers is not in the document]
Comment (Nitin-Kashyap) [reply be/src/vec/exec/format/parquet/parquet_bloom_reader.h:71]: fixed

File: be/src/vec/exec/format/parquet/parquet_bloom.h (hunk 1 above)
Comment (Nitin-Kashyap) [reply be/src/vec/exec/format/parquet/parquet_bloom.h:36]: fixed

**Pull request closed without merging**